### ISSUE TITLE
Fix 4.3 regression - Placeholders $product$ and $AssemblyGuid$ not replaced in contentfile through transformation

### ISSUE THREAD

**Issue by SabotageAndi** [NuGet/Home#5880]
## Details about Problem
I have a contentfile (.pp-file) with placeholders $product$ and $AssemblyGuid$. In VS 2015 and I think also before VS 2017 15.3.3 these are replaced with values during adding the package to the project.
When I add the package in the latest VS 2017 version, these placeholder are replaced with nothing.

Line of pp- file: `Settings projectName="$product$" projectId="{$AssemblyGuid$}" />`
Actual result: `<Settings projectName="" projectId="{}" />`
Expected result: `<Settings projectName="Name of the project" projectId="{a guid}" />`

Are these removed with the latest version, as I can not find them in the documentation of available properties https://msdn.microsoft.com/library/vslangproj.projectproperties_properties.aspx (linked from https://docs.microsoft.com/en-us/nuget/create-packages/source-and-config-file-transformations).

NuGet product used VS UI 
NuGet version (x.x.x.xxx): integrated one in Visual Studio
VS version: VS2017 - 15.3.3
OS version: Win 10 v1703 (10.0.15063)

Worked before? If so, with which NuGet version: 
It is still working in VS 2015 14.0.25431.01 Update 3

## Detailed repro steps so we can see the same problem

1. create new project 

2. add nuget package SpecRun.SpecFlow in version 1.6.0 from nuget.org

3. look at the added Default.srprofile file in line 3
actual: `<Settings projectName="" projectId="{}" />`
expected: `<Settings projectName="Name of the project" projectId="{a guid}" />`

**Comment by emgarten** [comment]
What type of project are you using? Is it a NETCore SDK project?

**Comment by SabotageAndi** [comment]
It is a classic class library project in the old csproj format.

**Comment by emgarten** [comment]
Here are the supported values for pp files: https://github.com/NuGet/NuGet.BuildTasks/blob/18aedb7eb0fc26012246af61e4df7cda4ab10026/src/Microsoft.NuGet.Build.Tasks/Microsoft.NuGet.targets#L140-L156

It doesn't look like AssemblyGuid is allowed.

**Comment by zhili1208** [comment]
I tried this package in VS 2015 with classic library and package.config, it's working, but on VS 2017, it doesn't work in same way. I think this is a regression.

**Comment by SabotageAndi** [comment]
@emgarten In your linked Microsoft.NuGet.targets file only 5 tokens are specified.
But the documentation (https://docs.microsoft.com/en-us/nuget/create-packages/source-and-config-file-transformations) says, there are more available:

> The $rootnamespace$ token is the most commonly used project property; all others are listed in the Project Properties (https://msdn.microsoft.com/library/vslangproj.projectproperties_properties.aspx) documentation on MSDN. Be mindful, of course, that some properties might be specific to the project type.

Is this really the code for replacing the tokens?
Additionally I am a little bit irritated, that this code is in a MSBuild target file and not in the C# code.

**Comment by emgarten** [comment]
@SabotageAndi that target is used for PackageReference/project.json, I was incorrect on pointing you towards that file. packages.config handles pp files differently and supports a wider range of properties.

**Comment by jainaashish** [comment]
Dupe of# https://github.com/dotnet/project-system/issues/2861

**Comment by SabotageAndi** [comment]
@jainaashish I don't think so, because I am using package.config and not PackageReference. It doesn't happen when you add the package to the project in VS.

**Comment by jainaashish** [comment]
@SabotageAndi `contentFiles` doesn't work with `packages.config`. For `packages.config`, you should use `Content` as explained here - https://docs.microsoft.com/en-us/nuget/schema/nuspec#including-content-files

**Comment by SabotageAndi** [comment]
@jainaashish They are in the content folder. I have no contentFiles in the package.

NuSpec:
```
<?xml version="1.0" encoding="utf-8"?>
<package xmlns="http://schemas.microsoft.com/packaging/2013/05/nuspec.xsd">
  <metadata>
    <id>SpecRun.SpecFlow.2-2-0</id>
    <version>1.6.2</version>
    <title>SpecFlow+Runner for SpecFlow 2.2</title>
    <authors>TechTalk</authors>
    <owners>TechTalk</owners>
    <requireLicenseAcceptance>false</requireLicenseAcceptance>
    <licenseUrl>http://www.specflow.org/plus/eula/</licenseUrl>
    <projectUrl>http://www.specflow.org/plus/runner</projectUrl>
    <iconUrl>http://www.specflow.org/media/specflow_plus_runner_nuget_icon.png</iconUrl>
    <description>SpecFlow+Runner - a smarter integration test runner for SpecFlow. Check the details on the website.</description>
    <summary>SpecFlow+Runner - a smarter integration test runner for SpecFlow. Check the details on the website.</summary>
    <copyright>Copyright © 2011-2017 TechTalk</copyright>
    <language>en-US</language>
    <tags>specrun specflow bdd integration tests runner</tags>
    <dependencies>
      <dependency id="SpecRun.Runner" version="[1.6.2]" />
      <dependency id="SpecFlow" version="[2.2.0, 2.3.0)" />
    </dependencies>
  </metadata>
</package>
```
Files:
![image](https://user-images.githubusercontent.com/859801/31093094-e20551c6-a7b0-11e7-8d28-f8897ce15f9d.png)

And as @zhili1208 said, the placeholder are working in VS2015.

**Comment by SabotageAndi** [comment]
I have this issue still with an actual VS2017 15.5.7. 
Should the fix already included in Visual Studio?

**Comment by rrelyea** [comment]
@SabotageAndi  - this is fixed in NuGet 4.6, which ships with Visual Studio 15.6 -- which is available as of this week.
